app/models/user.rb

```diff
@@ -478,6 +473,18 @@ def provider_id_for_audits
     account.try!(:provider_id_for_audits) || provider_account.try!(:provider_id_for_audits)
   end
 
+  def scope_services_with_permission
+    # TODO: this is probably wrong...
+    # how come if it does not have access_to_all_services but it can not use service_permissions,
+    # then we allow them all?!!
+    # but this is keeping the same behaviour that it previously had
+    if !has_access_to_all_services? && account.provider_can_use?(:service_permissions)
+      {id: member_permission_service_ids}
```

Comment:
Can we have this has as a real scope in Service? That looks strange using `id` but in fact referring to `Service#id`

Reply:
right, better 👍